src/code/z_parameter.c

```diff
@@ -4634,8 +4634,20 @@ void Interface_DrawClock(PlayState* play) {
                               R_THREE_DAY_CLOCK_SUN_MOON_CUTOFF * 4.0f);
 
             // determines the current hour
-            for (sp1C6 = 0; sp1C6 <= 24; sp1C6++) {
+            for (sp1C6 = 0; sp1C6 < ARRAY_COUNT(sThreeDayClockHours) - 1; sp1C6++) {
+                //! @bug In the original game, this loop iterates over an array of clock hour
+                // values to determine what the current hour is which is used to index into a
+                // texture pointer array. When the loop reaches the last value, the clock is
+                // actually equal to this value for a frame or two before it rolls over.
+                // Because this check is < and not <=, it will actually iterate past it by one
+                // due to the for loop terminating. This results in 25, which is OOB for the
+                // sThreeDayClockHourTextures[] read later. On console, this results in the hour
+                // disappearing for a frame or two between 11 changing to 12.
+#ifdef AVOID_UB
+                if (((void)0, gSaveContext.save.time) <= sThreeDayClockHours[sp1C6 + 1]) {
+#else
                 if (((void)0, gSaveContext.save.time) < sThreeDayClockHours[sp1C6 + 1]) {
+#endif
```

Comment:
As mentioned in discord, this is fixing a bug by changing behaviour, which isn't what we aim for with `AVOID_UB`